src/index.ts

```
@@ -1,34 +1,36 @@
-class User {
+interface User {
   name: string
   favoriteColor: string
   age: number
-  id: number
-
-  constructor(name: string, favoriteColor: string, age: number, id: number) {
-    this.name = name
-    this.favoriteColor = favoriteColor
-    this.age = age
-    this.id = id
-  }
+  id?: number
 }
 
 class UserAPI {
   users: Array<User>
+  currentId: number
 
   constructor() {
     this.users = []
+    this.currentId = 0
   }
 
   private assignId() {
-
+    this.currentId++
+    return this.currentId - 1
   }
 
   addUser(user: User) {
-
+    user.id = this.assignId()
```

what if the user is being added with an id that already exists?
in fact, if the user has an id, add should not work (its already added) ... in this case it should throw an error

also ... the added user should be returned